### PR TITLE
Added Standardized Abbreviations for Colleges, Majors, and Minors

### PR DESCRIPTION
**Pull request by jerrry1123**
### Summary 
- Added abbreviations for all colleges, majors, minors, and grad requirements
- added said abbreviations into the progress tracker toolcard

Remaining TODOs:

None

### Test Plan 

Test that the abbreviations are working correctly for both the export toolcard and the progress tracker toolcard.

### Notes 

There's a bug with the Applied Econ Minor because it's not in index.ts yet, we should add that soon.

### CODE REVIEW

File: src/requirements/decorated-requirements.json [hunk 1]
```diff
@@ -217243,97 +217283,12 @@
...
-          ]
-        }
-      ]
-    },
     "AEROSPACE": {
       "name": "Aerospace Engineering",
       "schools": [
         "EN"
       ],
+      "abbrev": "Aero Eng",
```
Comment (noschiff) [inline src/requirements/decorated-requirements.json:217291]: can we exclude the abbrev from the computed json like we do with advisors?

**Review by noschiff — CHANGES_REQUESTED** [review]

File: src/requirements/decorated-requirements.json (hunk 1 above)
Comment (jerrry1123) [reply src/requirements/decorated-requirements.json:217291]: it was included in req-gen which i had to run to pass the requirements check

File: src/requirements/types.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export type Major<R> = Readonly<{
   /** College requirements that have been "specialized" for this major */
   specializations?: readonly R[];
   advisors?: AdvisorGroup;
+  abbrev: string;
```
Comment (noschiff) [inline src/requirements/types.ts:29]: can we make this an optional field and exclude it from the generated JSON?

**Review by noschiff — CHANGES_REQUESTED** [review]

File: src/requirements/decorated-requirements.json (hunk 1 above)
Comment (noschiff) [reply src/requirements/decorated-requirements.json:217291]: yeah but we can exclude it from req-gen

File: src/components/Tools/ProgressTracker.vue [hunk 3]
```diff
@@ -124,6 +124,22 @@ export default defineComponent({
       }
     },
   },
+  methods: {
+    generateRequirementProgressString(req: string, reqGroup: string): string {
```
Comment (zachary0kent) [inline src/components/Tools/ProgressTracker.vue:128]: This name is too long. For example, the prefix `generate` doesn't convey any additional information.

File: src/components/Tools/ProgressTracker.vue [hunk 4]
```diff
@@ -124,6 +124,22 @@ export default defineComponent({
...
+        case 'College':
+          return getCollegeAbbrev(req) + ' ' + reqGroup + ' Requirements';
+        case 'Major':
+          return getMajorAbbrev(req) + ' ' + reqGroup + ' Requirements';
+        case 'Minor':
+          return getMinorAbbrev(req) + ' ' + reqGroup + ' Requirements';
+        case 'Grad':
+          return getGradAbbrev(req) + ' ' + reqGroup + ' Requirements';
+        default:
+          return '';
```
Comment (zachary0kent) [inline src/components/Tools/ProgressTracker.vue:139]: Can we refactor this big switch to use a function that returns the correct abbreviation given a requirement?

File: src/components/Tools/ProgressTracker.vue [hunk 5]
```diff
@@ -124,6 +124,22 @@ export default defineComponent({
       }
     },
   },
+  methods: {
+    generateRequirementProgressString(req: string, reqGroup: string): string {
+      switch (reqGroup) {
+        case 'College':
+          return getCollegeAbbrev(req) + ' ' + reqGroup + ' Requirements';
```
Comment (zachary0kent) [inline src/components/Tools/ProgressTracker.vue:131]: Use string interpolation instead

**Review by zachary0kent — CHANGES_REQUESTED** [review]
You are going to have to exclude the abbreviations from the generated JSON for the checks to pass. Check out how Vaish did this in #730.

**Review by zachary0kent — CHANGES_REQUESTED** [review]
We need to exclude the abbreviations from the generated requirements json.

**Review by zachary0kent — COMMENTED** [review]
There shouldn't be any diff on [src/requirements/decorated-requirements.json](https://github.com/cornell-dti/course-plan/pull/780/files#diff-38bfbe5fd0f55f8bc0869e1e39f5a52bfc0624e24709f1a69017aa7c448f3b1c)

**Comment by jerrry1123** [comment]
The only diff right now is moving the applied econ minor after the aerospace engineering minor instead of before, which might be because i switched the order in another file. should i move it back or leave it as is?

**Comment by zachary0kent** [comment]
> The only diff right now is moving the applied econ minor after the aerospace engineering minor instead of before, which might be because i switched the order in another file. should i move it back or leave it as is?

Oh I see, nevermind. It's alphabetized now which is better.

**Review by zachary0kent — CHANGES_REQUESTED** [review]
It seems the abbreviations appear correctly on the progress tracker toolcard, but not the exported schedule; see 'isst2'.
Screenshot 2023-02-24 at 2 01 56 PM

**Comment by jerrry1123** [comment]
Just added the codes to the export PDF! @zachary0kent

File: src/requirements/requirement-json-generator.ts [hunk 6]
```diff
@@ -355,22 +355,22 @@ const generateDecoratedRequirementsJson = (): DecoratedRequirementsJson => {
...
-    const { requirements, advisors, ...rest } = minorRequirement;
+    const { requirements, advisors, abbrev: abbr, ...rest } = minorRequirement;
     decoratedJson.minor[minorName] = {
       ...rest,
       requirements: decorateRequirements(requirements),
     };
   });
   Object.entries(grad).forEach(([gradName, gradRequirement]) => {
-    const { requirements, advisors, ...rest } = gradRequirement;
+    const { requirements, advisors, abbrev: abbr, ...rest } = gradRequirement;
```
Comment (zachary0kent) [inline src/requirements/requirement-json-generator.ts:373]:
```suggestion
    const { requirements, advisors, abbrev, ...rest } = gradRequirement;
```

**Review by zachary0kent — APPROVED** [review]
Looks good! Just left one small suggestion.

File: src/data/index.ts [hunk 7]
```diff
@@ -141,291 +141,339 @@ const json: RequirementsJson = {
...
       requirements: englishRequirements,
       advisors: englishAdvisors,
+      abbrev: 'English',
     },
     ENV: {
       name: 'Environmental Engineering',
       schools: ['EN'],
       requirements: envEngineeringRequirements,
       advisors: envEngineeringAdvisors,
+      abbrev: 'EnvEng',
```
Comment (noschiff) [inline src/data/index.ts:271]: is there any reason some abbreviations are space separated while others are PascalCase?

**Pull request merged**